### ISSUE TITLE
Allow for multiple solc versions in separate folders

### ISSUE THREAD

**Issue by justinjmoses** [juanfranblanco/vscode-solidity#172]
Related to @juanfranblanco's comment on https://github.com/juanfranblanco/vscode-solidity/issues/143 

> Let me understand the issue, so you have multiple folders (workspaces) and each one of them want to use a different version of solidity solc. At the moment the only way to override this, is to use a node package installation, but also this is always driven as well by the user settings.

> This is actually a very interesting change, but yes there are many points of failure as we move from user settings, folder settings to workspace settings.

> Ill do some research on this, but yeah a simple solution is that if you want to override a specific user setting you can use the npm package installation from the selected workspace folder.

Yes - we are in a situation where we are looking to support legacy contracts alongside modern contracts and need a way to have VSCode not complain like this: 

![image](https://user-images.githubusercontent.com/799038/80050785-9aa2ec80-84e4-11ea-9abf-721a293bd096.png)

Buidler supports multiple versions of solc, and this is a WIP branch: https://github.com/Synthetixio/synthetix/tree/adding-legacy-via-sources 

I'm really looking to see if I can stop my IDE from flagging these legacy sources as errors.

**Comment by juanfranblanco** [comment]
@justinjmoses  you can change the compiler version from your workspace using the settings and context menu.. 
![image](https://user-images.githubusercontent.com/562371/81044753-d35fa000-8eac-11ea-8ea0-1cd7e07af8d6.png)

**Comment by justinjmoses** [comment]
Right but I need two different versions within the one workspace...

**Comment by juanfranblanco** [comment]
The simplest solution is to open two instances of vscode each one with its own workspace settings. This way you can target specifically which version you want for each area.

**Comment by pilagod** [comment]
I happened to find out another workaround, you can turn off this setting in vscode:

`“solidity.enabledAsYouTypeCompilationErrorCheck”: false`

After that, vscode will not argue about the compiler version anymore. But this comes with a drawback that version specific syntax errors can only be found when you actually compile the files. 

For instance, in version `^0.6.0`, the `constructor` visibility needs to be specified. With this setting turning off, vscode won't remind you while editing code.

**Comment by juanfranblanco** [comment]
@pilagod yes, you will only get half of the functionality there. Much easier to open different folders.
Another option, is to change the settings on the fly, using the context menu.
Or as I do to work with optimism, mainly use a node setting configure to compile optimism, whilst working on latest all the time.

**Comment by zemse** [comment]
> The simplest solution is to open two instances of vscode each one with its own workspace settings.

Workaround is fine, but it'd be super if it just pick the lowest version that satisfies the pragma and this behavior turned on using some flag in `settings.json`.

**Comment by kryptoklob** [comment]
Echoing that this is still kind of a pain - it's very common to need different solidity compiler versions, sometimes even three different versions, within the same project.

**Comment by juanfranblanco** [comment]
So now that we have mono-repo support, I guess this could be achieved by adding another settings file with this or read it from somewhere like foundry etc

**Comment by kryptoklob** [comment]
> So now that we have mono-repo support, I guess this could be achieved by adding another settings file with this or read it from somewhere like foundry etc

Can you detail exactly how to do this? Can I just duplicate the settings file with different options?
